### PR TITLE
[Shadowlands][Disc Priest] Added Covenent Ability Boon of the Ascended.

### PR DESCRIPTION
**Pull request by Ogofo**
Hello,

this is my first PR on this proejct. I implemented the covenen module Boon of the Ascended for discipline priests. It seems to work, however I am unsure how I can further test and ensure the logic/parsing is correct. 

If you have any comments on this or the code alongside with recommendations, feel free to message me (either on discord: Ogofo, or here).

I wanted to thank @karlpralow, as he explained the basics to me :)

### CODE REVIEW

**Comment by Pewtro** [comment]
Thanks for wanting to contribute! 😄 

You should test it on a beta log that uses that ability to verify it works as it should. 

We have a CombatantInfoFaker file that you can fill info into to 'force' in the covenant being detected for the purpose of activating the module.

File: src/parser/priest/discipline/CHANGELOG.tsx [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+import { Ogofo, Oratio, Reglitch, Zerotorescue, niseko, Khadaj, blazyb, Adoraci, Tiphess, Putro } from 'CONTRIBUTORS';
 import SPELLS from 'common/SPELLS';
 import SpellLink from 'common/SpellLink';
 import ItemLink from 'common/ItemLink';
 import ITEMS from 'common/ITEMS';
 import { change, date } from 'common/changelog';
 
 export default [
+  change(date(2020, 10, 10), <>IMplementation of <SpellLink id={SPELLS.BOON_OF_THE_ASCENDED.id} />.</>, [Ogofo]),
   change(date(2020, 10, 3), <>Update <SpellLink id={SPELLS.POWER_WORD_SOLACE_TALENT.id} /> cooldown.</>, [Reglitch]),
```
Comment (Pewtro) [inline src/parser/priest/discipline/CHANGELOG.tsx:12]: Capital M should be lowercased

File: src/parser/priest/discipline/modules/shadowlands/covenants/BoonOfTheAscended.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  directHealing = 0;
+  damageDealt = 0; 
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasCovenant(COVENANTS.KYRIAN.id);
+
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.onHeal);
+  }
+
```
Comment (Pewtro) [inline src/parser/priest/discipline/modules/shadowlands/covenants/BoonOfTheAscended.tsx:36]: You should filter it down to blast, Nova and Eruption in the eventListener. 
```js
this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell([SPELLS.ASCENDED_BLAST, SPELLS.ASCENDED_NOVA, SPELLS.ASCENDED_ERUPTION])
```

File: src/parser/priest/discipline/modules/shadowlands/covenants/BoonOfTheAscended.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    
+    if (spellId === SPELLS.ASCENDED_NOVA.id) {
+      this.directHealing += event.amount + (event.absorbed || 0);
+      return;
+    }
+    if (spellId === SPELLS.ASCENDED_ERUPTION.id) {
+      this.directHealing += event.amount + (event.absorbed || 0);
+      return;
+    }
+  }
```
Comment (Pewtro) [inline src/parser/priest/discipline/modules/shadowlands/covenants/BoonOfTheAscended.tsx:49]: the entire onHeal should be rewritten after the eventlistener has been updated to be 
```js
onHeal(event: HealEvent) {
  if (isAtonement(event)) {

      const atonenementDamageEvent = this.atonementDamageSource.event;

      if (!atonenementDamageEvent) {
        return;
      }
      this.atonementHealing += event.amount + (event.absorbed || 0);
  } else {
    this.directHealing += event.amount + (event.absorbed || 0);
}
```

**Review by Pewtro — CHANGES_REQUESTED** [review]
Some things that you should change that will make it much more readable (and match the current codebase standards)

If I misunderstood how the atonement works, do say so - but from initial look it should work just fine.

**Comment by Ogofo** [comment]
@Pewtro Thanks for the help! Appreciated.
I implemented your suggestions. 

I tested the module with a [Disc-Priest beta log](https://www.warcraftlogs.com/reports/qrjaTbWKdHfx6Jcy#fight=25). It seems consistent. 

What's missing is that the icons are correctly displayed - however I cannot seem to find where to link/reference the correct icon art. Could you point me there?

**Review by Pewtro — APPROVED** [review]

**Comment by Pewtro** [comment]
So long as the icon is correctly added in the spell object (it looks to be), it'll work just fine when wowhead switches over. 
(They might exist only on shadowlands.wowhead.com)

**Comment by Pewtro** [comment]
Thanks for the PR!

**Pull request merged**